### ISSUE TITLE
Cannot "esy bootstrap" on Arch Linux.

### ISSUE THREAD

**Issue by morteify** [onivim/oni2#945]
I was following [oni2 build guide](https://onivim.github.io/docs/for-developers/building) and after passing first step of "esy install", I got stuck at "esy bootstrap", which returns following error: 
```
/home/mweglarz/sources/oni2/_esy/default/pnp.js:2754
    for (const path of paths) {
                       ^

TypeError: paths is not iterable
    at Function.Module._findPath (/home/mweglarz/sources/oni2/_esy/default/pnp.js:2754:24)
    at resolveMainPath (internal/bootstrap/pre_execution.js:447:28)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1138:24)
    at internal/main/run_main_module.js:16:11
```
I've installed every Arch Linux requirement listed for oni2 and [revery](https://github.com/revery-ui/revery/wiki/Building-&-Installing) but cannot simply get it running.

**Comment by bryphe** [comment]
Interesting, seems like the same callstack as this issue: https://github.com/esy/esy/issues/1010

Which version of `esy` are you running, @morteify ?

**Comment by morteify** [comment]
@bryphe 

- esy: 0.5.8
- node: 13.1.0
- linux kernel: 5.3.11-arch1-1

**Comment by snadon** [comment]
Getting the same problem on mac:

- esy: 0.5.6
- node: 13.1.0
- catalina: 10.15.1

**Comment by ifreund** [comment]
I get the same error as well.  On arch with esy 0.5.8 and node 13.1.0.

**Comment by tiensonqin** [comment]
esy 0.5.8 and node v12.13.0 works for me.

**Comment by CrossR** [comment]
Just to confirm the pattern that was already becoming obvious here:

On Mac with esy 0.5.8 and node 12.12.0 I had no issues. I then updated to move to node 13.1.0 and I got this issue, so it looks to be related to the node version.

From a quick poke around, the difference is that now `paths` isn't being set. Ie `console.log(paths)` returns `null` on 13.

**Comment by bryphe** [comment]
Thanks for the details, @CrossR !

Indeed - looks like `esy` generates a `pnp.js` file from here:
https://github.com/esy/esy/blob/cee0381a84c14c4d68d3f1d8a6a5f4d46fecb363/esy-install/PnpJs.re#L944

...and in that spot, it 'monkey-patches' the `Module._findPath` module resolution logic. It seems that, in Node 13+, that `paths` can now be `null`.

Looking at the original implementation: https://github.com/nodejs/node/blob/efce655c0f1671d0e86b5c89092ac93db983ef94/lib/internal/modules/cjs/loader.js#L617 it seems it checks if the path is absolute, if it is, it just uses `['']` for `paths`- otherwise it validates it is non-null and an array. Seems like we might need to bring some of that logic over to esy.

**Comment by morteify** [comment]
@CrossR @bryphe Can confirm that after downgrade of node the issue is gone and everything works properly

**Comment by imbsky** [comment]
https://github.com/esy/esy/pull/1018 should fix this issue.

**Comment by imbsky** [comment]
`@esy-nightly/esy` should work without this error because https://github.com/esy/esy/pull/1018 has been merged.

**Comment by Et7f3** [comment]
It seem to have this in multiple place.
```
/home/elie/e/_esy/default/pnp.js:694
    for (const path of paths) {
                       ^

TypeError: paths is not iterable
    at Function.Module._findPath (/home/elie/e/_esy/default/pnp.js:694:24)
    at resolveMainPath (internal/modules/run_main.js:12:25)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:61:24)
    at internal/main/run_main_module.js:17:47```
@imbsky

**Comment by imbsky** [comment]
@Et7f3 Try this. This should be done very quickly because you have cache.

```
rm -r _esy esy.lock node_modules
esy
```

**Comment by imbsky** [comment]
Oh, probably delete `_esy` is just enough.

**Comment by Et7f3** [comment]
Ok :ok_hand: Work well. Thanks.

**Comment by imbsky** [comment]
I'm glad to hear that it worked without any errors!

**Comment by bryphe** [comment]
Thank you @imbsky for fixing esy to handle this (in https://github.com/esy/esy/pull/1018) - really appreciate it!

**Comment by bryphe** [comment]
Once the new version of `esy` is released - I'll update our CI and build steps to pick it up, and then we can close this out.

**Comment by imbsky** [comment]
@bryphe Anytime!

**Comment by Thewessen** [comment]
Is this also fixed for macOS? Still have this issue with fresh master pull today. Any help?

esy: 0.5.8
node: 13.2.0
catalina: 10.15.2

```
/Users/samuelthewessen/oni2/_esy/default/pnp.js:2782
    for (const path of paths) {
                       ^

TypeError: paths is not iterable
    at Function.Module._findPath (/Users/samuelthewessen/oni2/_esy/default/pnp.js:2782:24)
```

**Comment by imbsky** [comment]
@Thewessen Install `@esy-nightly/esy` instead of `esy`. Then remove the `_esy` directory and execute esy command again. (and this is all in this thread...)

**Comment by Thewessen** [comment]
Now I understand the post: `@esy-nightly/esy` should fix this...  ; | Thanks @imbsky, working great! Sorry for my duplicate question

**Comment by imbsky** [comment]
No worries. but the current documentation is somewhat unfriendly. It's probably not a bad idea to add a note to a document.

**Comment by imbsky** [comment]
I opened a PR.

**Comment by imbsky** [comment]
Related PR: https://github.com/onivim/oni2/pull/1200

**Comment by bryphe** [comment]
Fixed now in esy@0.6.0! #1200 updates our scripts to use esy@0.6.0 on CI, and I've updated our documentation to prescribe using esy@0.6.0 for the build steps

Thanks @CrossR @imbsky @morteify  for the help (and @andreypopp for fixing / releasing esy@0.6.0). Will close now
